clients/auth/auth.go

```diff
@@ -37,43 +39,71 @@ type printParams struct {
 	tokens  []token
 }
 
+type ResourcePermission struct {
+	Name    string
+	Read    bool
+	Write   bool
+	IsCloud bool
+	IsOss   bool
+}
+
 type CreateParams struct {
 	clients.OrgParams
 	User        string
 	Description string
 
-	WriteUserPermission bool
-	ReadUserPermission  bool
-
-	WriteBucketsPermission bool
-	ReadBucketsPermission  bool
+	ResourcePermissions []*ResourcePermission
```

Comment:
Why `[]*ResourcePermission` instead of `[]ResourcePermission`?

Reply:
`append` does copies but I give away pointers to the internal state (for the argument setters).